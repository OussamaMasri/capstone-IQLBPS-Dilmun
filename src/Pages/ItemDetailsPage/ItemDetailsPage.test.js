import React from 'react';
import { render, cleanup } from '@testing-library/react';
import { act } from 'react-dom/test-utils';
import { Provider } from 'react-redux';
import ItemDetailsPage from './ItemDetailsPage';
import { store } from '../../app/store';

const MOCK_ITEM_DATA = {
  title: 'Title',
  file: 'https://cdn.discordapp.com/attachments/1031834305703460906/1041710013992947812/image.png',
  description: 'Description',
  price: 110,
  type: 'Item Type',
  categories: ['Men'],
  owner: {
    email: 'email',
  },
};

afterEach(cleanup);

test('Sort the Products', async () => {
  const tree = render(
    <Provider store={store}>
      <ItemDetailsPage item={MOCK_ITEM_DATA} />
    </Provider>
  );
  await act(() => tree);
  expect(tree).toMatchSnapshot();
});
