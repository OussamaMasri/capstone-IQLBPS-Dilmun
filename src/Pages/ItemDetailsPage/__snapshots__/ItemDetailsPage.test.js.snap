// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sort the Products 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="flex flex-col md:flex-row text-background p-10 md:p-20 bg-background w-full"
        data-testid="item-details-page"
      >
        <div
          class="w-full lg:w-1/2"
        >
          <img
            class="max-h-full min-h-[600px] w-full object-cover md:object-fill	"
          />
        </div>
        <div
          class="w-full lg:w-1/2 md:ml-auto max-h-full md:mr-auto flex flex-col gap-7 rounded pt-5 px-5 bg-gradient-to-tl from-tertiary to-primary justify-between py-10 min-h-[600px]"
        >
          <div
            class="flex flex-col gap-7"
          >
            <div>
              <h2
                class="font-bold text-4xl pb-4"
              />
              <div
                class=""
              >
                <span
                  class="text-lg font-medium text-background bg-tertiary bg-opacity-50 rounded-full px-2 py-1 mr-2 shadow-md"
                />
              </div>
            </div>
            <span
              class="flex gap-1"
            >
              <p
                class="font-bold"
              >
                items_details.item_type
              </p>
            </span>
            <span
              class="flex gap-1"
            >
              <p
                class="font-bold"
              >
                items_details.price
              </p>
               $
            </span>
            <span
              class="flex gap-1"
            >
              <p
                class="font-bold"
              >
                items_details.address
              </p>
              undefined, undefined
            </span>
            <div>
              <span
                class="font-bold text-xl"
              >
                items_details.description
              </span>
              <p
                class="text-lg"
              />
            </div>
          </div>
          <div
            class="w-full self-end"
          >
            <a
              class="w-full"
              href="mailto:undefined"
              rel="noopener noreferrer"
              target="_blank"
            >
              <button
                class="bg-tertiary px-4 py-4 w-full text-white font-semibold rounded-lg hover:shadow-lg hover:bg-secondary transition duration-500 cursor-pointer lg:w-full"
                type="submit"
              >
                items_details.email_owner
              </button>
            </a>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="flex flex-col md:flex-row text-background p-10 md:p-20 bg-background w-full"
      data-testid="item-details-page"
    >
      <div
        class="w-full lg:w-1/2"
      >
        <img
          class="max-h-full min-h-[600px] w-full object-cover md:object-fill	"
        />
      </div>
      <div
        class="w-full lg:w-1/2 md:ml-auto max-h-full md:mr-auto flex flex-col gap-7 rounded pt-5 px-5 bg-gradient-to-tl from-tertiary to-primary justify-between py-10 min-h-[600px]"
      >
        <div
          class="flex flex-col gap-7"
        >
          <div>
            <h2
              class="font-bold text-4xl pb-4"
            />
            <div
              class=""
            >
              <span
                class="text-lg font-medium text-background bg-tertiary bg-opacity-50 rounded-full px-2 py-1 mr-2 shadow-md"
              />
            </div>
          </div>
          <span
            class="flex gap-1"
          >
            <p
              class="font-bold"
            >
              items_details.item_type
            </p>
          </span>
          <span
            class="flex gap-1"
          >
            <p
              class="font-bold"
            >
              items_details.price
            </p>
             $
          </span>
          <span
            class="flex gap-1"
          >
            <p
              class="font-bold"
            >
              items_details.address
            </p>
            undefined, undefined
          </span>
          <div>
            <span
              class="font-bold text-xl"
            >
              items_details.description
            </span>
            <p
              class="text-lg"
            />
          </div>
        </div>
        <div
          class="w-full self-end"
        >
          <a
            class="w-full"
            href="mailto:undefined"
            rel="noopener noreferrer"
            target="_blank"
          >
            <button
              class="bg-tertiary px-4 py-4 w-full text-white font-semibold rounded-lg hover:shadow-lg hover:bg-secondary transition duration-500 cursor-pointer lg:w-full"
              type="submit"
            >
              items_details.email_owner
            </button>
          </a>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
